src/GovUk.Education.ExploreEducationStatistics.Data.Model.Tests/Repository/FootnoteRepositoryTests.cs

```diff
@@ -1,16 +1,1222 @@
+            {
+                await context.Subject.AddRangeAsync(subject1, subject2);
+                await context.Filter.AddRangeAsync(subject1Filter1, subject1Filter2, subject1Filter3);
+                await context.IndicatorGroup.AddAsync(subject1IndicatorGroup1);
+                await context.Release.AddAsync(release);
+                await context.ReleaseSubject.AddRangeAsync(releaseSubjects);
+                await context.SaveChangesAsync();
+            }
+
+            await using (var context = InMemoryStatisticsDbContext(contextId))
+            {
+                var repository = BuildFootnoteRepository(context);
+
+                //  Get the footnotes applying to subject 1 or any of its filter items or indicators
+                var filter1Group1Item1Id = subject1Filter1.FilterGroups[0].FilterItems[0].Id;
+                var filter2Group1Item1Id = subject1Filter2.FilterGroups[0].FilterItems[0].Id;
+                var filter3Group1Item1Id = subject1Filter3.FilterGroups[0].FilterItems[0].Id;
+                var indicatorGroup1Item1Id = subject1IndicatorGroup1.Indicators[0].Id;
+
+                var results = await repository.GetFilteredFootnotes(
+                    releaseId: release.Id,
+                    subjectId: subject1.Id,
+                    filterItemIds: ListOf(filter1Group1Item1Id, filter2Group1Item1Id, filter3Group1Item1Id),
+                    indicatorIds: ListOf(indicatorGroup1Item1Id));
+
+                // Check that only footnotes applying to subject 1 or any of its filter items or indicators are returned
+                Assert.Equal(6, results.Count);
+
+                // Footnote applies to the requested subject
+                Assert.Equal(releaseFootnotes[0].FootnoteId, results[0].Id);
+                Assert.Equal("Footnote 1", results[0].Content);
+
+                // Footnote applies to a requested filter item via its filter
+                Assert.Equal(releaseFootnotes[1].FootnoteId, results[1].Id);
+                Assert.Equal("Footnote 2", results[1].Content);
+
+                // Footnote applies to a requested filter item via its filter group
+                Assert.Equal(releaseFootnotes[2].FootnoteId, results[2].Id);
+                Assert.Equal("Footnote 3", results[2].Content);
+
+                // Footnote applies to a requested filter item
+                Assert.Equal(releaseFootnotes[3].FootnoteId, results[3].Id);
+                Assert.Equal("Footnote 4", results[3].Content);
+
+                // Footnote applies to a requested indicator 
+                Assert.Equal(releaseFootnotes[4].FootnoteId, results[4].Id);
+                Assert.Equal("Footnote 5", results[4].Content);
+
+                // Footnote applies to a various criteria related to the requested filter items and indicators
+                Assert.Equal(releaseFootnotes[5].FootnoteId, results[5].Id);
+                Assert.Equal("Footnote 6", results[5].Content);
+            }
+        }
+
```

Comment:
Might be a good idea to add other test cases for:

- returning footnotes that match only one filter item / filter group / filter (from a selection)
- returning footnotes that match only one indicator (from a selection)

In the above test, we're providing all the criteria, and returning all the footnotes. This means we can't be absolutely certain that the method isn't just returning _all_ the subject's footnotes by default.

Some of the other test cases semi-cover this, but it'd be good to have dedicated test cases for this it to be more airtight.

Reply:
I've added these tests now, thanks!